invenio_oauthclient/utils.py

```diff
@@ -103,14 +103,87 @@ def oauth_authenticate(client_id, user, require_existing_link=False):
     return False
 
 
-def oauth_register(form):
+def filter_user_info(user_info, precedence_mask=None):
+    """Filter the user info dictionary according to the precedence mask.
+
+    If the precedence_mask keyword argument is not specified, the value
+    from the configuration is used.
+    This should be the normal call convention of this function.
+
+    :param user_info: The user info dictionary.
+    :param precedence_mask: The precedence mask to use.
+    """
+    if precedence_mask is None:
+        precedence_mask = current_app.config.get(
+            "OAUTHCLIENT_USER_INFO_PRECEDENCE_MASK",
+            {"email": True},
+        )
+
+    # for each of the user info values, check if they are supposed
+    # to take precedence over user input (as per precedence mask)
+    for key, user_info_value in list(user_info.items()):
+        precedence_value = precedence_mask.get(key, False)
+        info_val_dict = isinstance(user_info_value, dict)
+        prec_val_dict = isinstance(precedence_value, dict)
+
+        if info_val_dict and prec_val_dict:
+            # if both values in the mask and user_info are dicts,
+            # investigate deeper
+            filter_user_info(user_info_value, precedence_mask[key])
+
+        elif prec_val_dict:
+            # the precedence dictionary says it's a dict, but
+            # it's actually a value... inconsistent!
+            user_info.pop(key, None)
+
+        elif not precedence_value:
+            user_info.pop(key, None)
+
+
+def patch_dictionary(orig_dict, patch_dict):
+    """Replace the fields mentioned in the patch, while leaving others as is.
+
+    Note: The first argument's content will be changed during the process.
+
+    :param orig_dict: A dictionary.
+    :param patch_dict: The dictionary whose values should take precedence.
+    """
+    for key in patch_dict.keys():
+        val = patch_dict[key]
+        orig_val = orig_dict.get(key)
+
+        if isinstance(val, dict) and isinstance(orig_val, dict):
+            patch_dictionary(orig_dict[key], val)
+        else:
+            orig_dict[key] = val
+
+
+def remove_csrf_tokens(user_data):
+    """Remove CSRF tokens from the user data."""
+    user_data.pop("csrf_token", None)
+    for key, value in list(user_data.items()):
+        if isinstance(value, dict):
+            remove_csrf_tokens(value)
```

Comment:
removing the `csrf_token` properties from the user data before passing it to `register_user` avoids the unexpected keyword argument error that i mentioned at some point